### DISCUSSION TITLE
Are Windows named pipes available for WSL2?

### DISCUSSION THREAD

**Discussion by MathijsNL** [microsoft/WSL#5919] [General] [answered]
I would like to use named pipes that are created under Windows to be available for programs running under WSL2. So far I have had no luck finding anything about it.

Is this option available, and if not will it be implemented in the future?

**Comment by Biswa96** [comment, marked as answer]
Socket is the documented way to bridge Win32 and WSL side. Pipe is hidden.

**Reply by MathijsNL** [reply]
Thanks for your reply!

For other people that are looking for this, the link below gave me some good info about how to get this working.

[https://devblogs.microsoft.com/commandline/windowswsl-interop-with-af_unix/](https://devblogs.microsoft.com/commandline/windowswsl-interop-with-af_unix/)
